src/NormalizedProgress.cpp

```diff
@@ -4,56 +4,29 @@
 #include "GenericProgressCallback.h"
 
 //system
+#include <atomic>
 #include <cassert>
 
-#ifdef USE_QT
-
-//we use Qt for the atomic counter
-#include <QAtomicInt>
-
-class AtomicCounter : public QAtomicInt
-{
-};
-
-#else
-
-//we use a fake QAtomicInt
-
-//! Fake QAtomicInt
-/** \warning Not thread safe!
-**/
-class AtomicCounter
-{
-public:
-	AtomicCounter() : m_value(0) {}
-	inline int load() { return m_value; }
-	inline void store(int value) { m_value = value; }
-	inline int fetchAndAddRelaxed(int add) { int original = m_value; m_value += add; return original; }
-	int m_value;
-};
-
-#endif
-
 using namespace CCLib;
 
+// Use a class "wrapper" to avoid having to include <atomic> in header
+class CCLib::AtomicCounter : public std::atomic_uint {};
```

Comment:
Actually we could use the STL atomic counter directly (no need for a fake interface and a pointer) as the aim here was to compile CCCoreLib without Qt, and not without thread safety.

If there was an equivalent to the Qt Concurrent 'map' feature used in DgmOctree (QtConcurrentMap), we could actually remove the whole dependency to Qt I believe.

Reply:
> we could use the STL atomic counter directly

We could, but as I point out in the comment I don't want to include `<atomic>` in the header. (Comments in code are useless 😄)

We should probably use PIMPL throughout, but that's not a small change!

> we could actually remove the whole dependency to Qt I believe.

Yes - this would be a goal. We went down this road years ago and did not find a solution. I'll add an issue for it here.